lib/invoker.js

```diff
@@ -47,9 +48,32 @@ module.exports = function invoker(func) {
 
     // Check for user defined headers
     if (typeof payload.response.headers === 'object') {
-      Object.assign(payload.headers, payload.response.headers);
+      const headers = {};
+      // normalize the headers as lowercase
+      for (const header in payload.response.headers) {
+        headers[header.toLocaleLowerCase()] = payload.response.headers[header];
+      }
+      payload.headers = { ...payload.headers, ...headers };
       delete payload.response.headers;
     }
+
+    // If the response is a CloudEvent, we need to convert it
+    // to a Message first and respond with the headers/body
+    if (payload.response instanceof CloudEvent) {
+      try {
+        const message = HTTP.binary(payload.response);
+        payload.headers = {...payload.headers, ...message.headers};
+        payload.response = message.body;  
+      } catch (e) {
+        console.error(e);
```

Comment:
Shouldn't we set `500` status code here, or is it handled somewhere else?

Reply:
Ahh good catch! Fixed.